src/components/elements/table/GenericTable.tsx

```diff
@@ -212,6 +214,11 @@ const GenericTable = <T extends { id: string }>({
 
   // initially undefined so we can early return and avoid state flicker
   const [selected, setSelected] = useState<string[]>();
+  const isSelectControlled = selectedProp !== undefined;
+  const selectedValue = useMemo(
+    () => (isSelectControlled ? selectedProp : selected || []),
+    [isSelectControlled, selected, selectedProp]
+  );
```

Comment:
to discuss: refactor option to move selection logic into a hook? there's a lot going on. I do agree we need the controlled selection behavior you added.